packages/truffle-codec/lib/decode/value.ts

```diff
@@ -36,142 +40,238 @@ export default function* decodeValue(dataType: Types.Type, pointer: DataPointer,
 
     case "bool": {
       if(!checkPaddingLeft(bytes, 1)) {
-        let error = new Errors.BoolPaddingError(CodecUtils.Conversion.toHexString(bytes));
+        let error = {
+          kind: "BoolPaddingError" as "BoolPaddingError",
+          raw: CodecUtils.Conversion.toHexString(bytes)
+        };
```

Comment:
you know, you could make helper functions for these, so you don't have to deal with all these object literals all the time.

Reply:
(that'd be the redux-y way to do it)

Reply:
Hm, I guess.  Not clear it'd save that much trouble though.  I don't really want to bother at this point, and as I said I'm not actually convinced it'd help...

Reply:
Yeah I'm not about to suggest doing it now that you've done all this work making them objects. In the future, maybe it's worth doing if we're doing other refactoring work.

The reason for doing it is really just for clarity, esp. if you're looking at the helper functions inside an IDE that gives you jump-to-definition functionality (or whenever GH finally gives us that feature on this repo)